File: frontend/src/types/staff.ts
export interface Staff {
  id: string;
  facility_id: string;
  first_name: string;
  last_name: string;
  dni: string | null;
  phone: string | null;
  email: string | null;
  position: string | null;
  hire_date: string | null;
  is_active: boolean;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface StaffCreate {
  facility_id: string;
  first_name: string;
  last_name: string;
  dni?: string;
  phone?: string;
  email?: string;
  position?: string;
  hire_date?: string;
  notes?: string;
}

export interface StaffUpdate {
  first_name?: string;
  last_name?: string;
  dni?: string;
  phone?: string;
  email?: string;
  position?: string;
  hire_date?: string;
  is_active?: boolean;
  notes?: string;
}

File: backend/app/schemas/staff.py
from pydantic import BaseModel
from uuid import UUID
from datetime import date, datetime
from typing import Optional


class StaffCreate(BaseModel):
    facility_id: UUID
    first_name: str
    last_name: str
    dni: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    position: Optional[str] = None
    hire_date: Optional[date] = None
    notes: Optional[str] = None


class StaffUpdate(BaseModel):
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    dni: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    position: Optional[str] = None
    hire_date: Optional[date] = None
    is_active: Optional[bool] = None
    notes: Optional[str] = None


class StaffResponse(BaseModel):
    id: UUID
    facility_id: UUID
    first_name: str
    last_name: str
    dni: Optional[str]
    phone: Optional[str]
    email: Optional[str]
    position: Optional[str]
    hire_date: Optional[date]
    is_active: bool
    notes: Optional[str]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
